addons/website_blog/controllers/main.py

```diff
@@ -149,6 +149,9 @@ def _prepare_blog_values(self, blogs, blog=False, date_begin=False, date_end=Fal
     ], type='http', auth="public", website=True)
     def blog(self, blog=None, tag=None, page=1, **opt):
         Blog = request.env['blog.blog']
+        # TODO: remove in master
+        if isinstance(blog, str):
+            blog = Blog.search([('id', '=', unslug_object_str(blog)['id'])], limit=1)
```

Comment:
- Why no comments to explain what this is about ?
- `.search([('id', '=', x)], limit=1)`... surely you must know about `browse` ? :grimacing: